### PR TITLE
[6_2_X][TIMOB-24814] Label regression

### PR DESCRIPTION
**Pull request by infosia**
[TIMOB-24814](https://jira.appcelerator.org/browse/TIMOB-24814)

```js
var win = Ti.UI.createWindow({ backgroundColor: 'green' }),
    view0 = Ti.UI.createView({
        top: 0,
        left: 0,
        height: Ti.UI.FILL,
        width: '50%',
        backgroundColor: 'red'
    }),
    view1 = Ti.UI.createView({
        top: 0,
        right: 0,
        height: 300,
        width: 300,
        backgroundColor: 'blue'
    }),
    label0 = Ti.UI.createLabel({
        width: Ti.UI.SIZE,
        height: Ti.UI.SIZE,
        color: "#000",
        font: {
            fontSize: 12
        },
        text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?'
    }),
    label1 = Ti.UI.createLabel({
        width: Ti.UI.SIZE,
        height: Ti.UI.SIZE,
        color: "#000",
        font: {
            fontSize: 12
        },
        text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?'
    });

view0.add(label0);
view1.add(label1);

win.add(view0);
win.add(view1);

win.open();
```

Also make sure [TIMOB-24831](https://jira.appcelerator.org/browse/TIMOB-24831) is met too.

```js
var _window = Ti.UI.createWindow();
var lbl = Ti.UI.createLabel({
    text: 'number1. This is some long text. 2. This is some long text. 3. This is some long text. 4. This is some long text. 5. This is some long text. 6. This is some long text. 7. This is some long text. 8. This is some long text. 9. This is some long text. 10. This is some long text. 11. This is some long text. 12. This is some long text. 13. This is some long text. 14. This is some long text. 15. This is some long text. 16. This is some long text. 17. This is some long text. 18. This is some long text.',
    right: 10,
    left: 10,
    top: 10,
    bottom: 10,
    borderWidth: 1,
    borderColor: 'blue'
});
_window.add(lbl);
_window.open();
```

### CODE REVIEW

File: Source/UI/include/TitaniumWindows/UI/Label.hpp [hunk 1]
```diff
@@ -64,6 +64,10 @@ namespace TitaniumWindows
 			Windows::UI::Xaml::Controls::TextBlock^ label__;
 
 			bool need_measure__ { false };
+#pragma warning(push)
+#pragma warning(disable : 4251)
+			Titanium::LayoutEngine::Rect oldRect__;
```
Comment (garymathews) [inline Source/UI/include/TitaniumWindows/UI/Label.hpp]: Maybe call it `previousSize__` for clarity?

**Comment by ewanharris** [comment]
@infosia This isn't a proper FR, but I was trying this change with a layout problem I saw while running 6.2.0 tests, and the following crashes for me. It looks to be tied to setting top and bottom at the same time

```js
var win = Ti.UI.createWindow();
var data = [];
data[0] = Ti.UI.createTableViewRow({hasChild:true,height:'auto'});
data[0].add(Ti.UI.createLabel({
	text:'There should be 18 sentences: 1. This is some long text. 2. This is some long text. 3. This is some long text. 4. This is some long text. 5. This is some long text. 6. This is some long text. 7. This is some long text. 8. This is some long text. 9. This is some long text. 10. This is some long text. 11. This is some long text. 12. This is some long text. 13. This is some long text. 14. This is some long text. 15. This is some long text. 16. This is some long text. 17. This is some long text. 18. This is some long text.',
	height:'auto',
	width:'auto',
	left:10,
	right:50,
	top:10,
	bottom:10
}));
var tableview = Titanium.UI.createTableView({
	data:data,
	minRowHeight:80
});
win.add(tableview);
win.open();
````

**Comment by garymathews** [comment]
@infosia The desired size of Label was ignoring `right` and `bottom` padding. I was going to comment the changes here but decided it was easier to push a [commit](https://github.com/appcelerator/titanium_mobile_windows/pull/1099/commits/8686e1ac8c86817691bc2dcf90228ac0449351f6) to #1099 

It should fix Ewans test case too 👍

**Comment by infosia** [comment]
@garymathews Awesome, looks good to me. Merged to this branch too. Thanks 👍

**Comment by ewanharris** [comment]
@infosia @garymathews Here's what the failures on jenkins are for failure `TIMOB-23372 #4 ` on Jenkins, if you look closely the yellow view height grows slightly with the fix, it's only noticeable for me when switching between the two

### With fix
With fix

### Without fix

With fix

**Pull request merged**